### PR TITLE
verapdf 1.20.3 (new formula)

### PR DESCRIPTION
**Pull request by alexreg**
Note that the results of the audit are the following.

```
verapdf-apps:                                                                                       
  * Non-executables were installed to "/opt/homebrew/opt/verapdf-apps/bin".                         
    The offending files are:                                                                        
      /opt/homebrew/opt/verapdf-apps/bin/greenfield-apps-1.21.0-SNAPSHOT.jar                        
  * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)                    
Error: 2 problems in 1 formula detected                                                             
```

However, the first point should be countered by the fact that the jar file is required by all the binaries in `bin/`, and this is the manner in which the software itself installs it. The second point should be countered by the fact that the veraPDF/veraPDF-library repo easily satisfies the notability constraint, and is the primary repo for all work on veraPDF (veraPDF-apps does *not* have its own issue tracker, for example), and apparently many people interested in veraPDF as a whole have only starred this repo.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for thexsame formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by alexreg** [comment]
No idea why this is failing, as the local build works on arm64.

File: Formula/verapdf-apps.rb
Comment (carlocab) [inline Formula/verapdf-apps.rb:34]: This looks rather difficult to maintain. Is there not something upstream we can use? Or can this be upstreamed instead?

File: Formula/verapdf-apps.rb
Comment (carlocab) [inline Formula/verapdf-apps.rb:42]:
```suggestion
    with_env(VERAPDF: "#{bin}/verapdf", NO_CD: "1") do
      system prefix/"tests/exist-status.sh"
    end
```

**Review by carlocab — COMMENTED** [review]
> However, the first point should be countered by the fact that the jar file is required by all the binaries in `bin/`, and this is the manner in which the software itself installs it.

That's not a good reason. `bin` should not contain non-executables. If this needs the `jar` in the same directory, install everything into `libexec` and then create a symlink or exec script in `bin` as appropriate.

File: Formula/verapdf-apps.rb
Comment (alexreg) [reply Formula/verapdf-apps.rb:34]: There's an easier way to change the install prefix, but there's no easier way to change the pack options — I already discussed this with the developers.

**Comment by alexreg** [comment]
> That's not a good reason. `bin` should not contain non-executables. If this needs the `jar` in the same directory, install everything into `libexec` and then create a symlink or exec script in `bin` as appropriate.

Sure, I'll try to implement that.

**Comment by alexreg** [comment]
Hmm, annoyingly the installer puts the script in the prefix dir itself, and those scripts are hardcoded to expect the jar file in the bin dir. Any suggestions?

File: Formula/verapdf-apps.rb
Comment (carlocab) [inline Formula/verapdf-apps.rb:19]:
```suggestion
    system "java", "-DINSTALL_PATH=#{libexec}", "-jar", installer_file, "-options-system"
    bin.install_symlink "#{libexec}/verapdf", "#{libexec}/verapdf-gui"
```

Comment (alexreg) [reply Formula/verapdf-apps.rb:19]: That's one way of doing it, yep!

File: Formula/verapdf-apps.rb
Comment (carlocab) [inline Formula/verapdf-apps.rb:10]:
```suggestion
  depends_on "openjdk"
```
Looks like you need this at runtime.

File: Formula/verapdf-apps.rb
Comment (carlocab) [inline Formula/verapdf-apps.rb:18]:
```suggestion
    bin.install libexec/"verapdf", libexec/"verapdf-gui"
    bin.env_script_all_files libexec, JAVA_HOME: Language::Java.overridable_java_home_env
```

**Comment by alexreg** [comment]
Thanks for the prompt reviews, @carlocab. Hopefully everything is in order now.

**Comment by alexreg** [comment]
Actually, I'm afraid the program fails to run since I incorporated your latest review suggestions.
```
Error: JAVA_HOME is not defined correctly.
  We cannot execute {:JAVA_HOME=>/opt/homebrew/opt/openjdk/libexec/openjdk.jdk/Contents/Home}/bin/java
```

File: Formula/verapdf-apps.rb
Comment (carlocab) [inline Formula/verapdf-apps.rb:19]:
```suggestion
    bin.env_script_all_files libexec, Language::Java.overridable_java_home_env
```
Oops, my mistake

Comment (alexreg) [reply Formula/verapdf-apps.rb:19]: Is this better than the following?
```
bin.env_script_all_files libexec, JAVA_HOME: Formula["openjdk"].opt_prefix
```
(I see that in various other formulas.)

Comment (carlocab) [reply Formula/verapdf-apps.rb]: Yes, it's better.

Comment (alexreg) [reply Formula/verapdf-apps.rb]: Pushed already. It looks like the only issue now is the exit code for the out-of-memory test, which is wrong on arm64 for some reason.

**Comment by alexreg** [comment]
Just waiting on a new verapdf release that fixes the exit code for OOM issues...

**Comment by alexreg** [comment]
@carlocab I believe all tests pass now, so this should be ready to merge.

**Comment by alexreg** [comment]
The only outstanding issue is the audit one, mentioned in the original post, which should probably be bypassed.

**Comment by alexreg** [comment]
Anyone?

File: Formula/verapdf-apps.rb
Comment (carlocab) [inline Formula/verapdf-apps.rb:1]: One last question: other packagers (e.g. Arch) seem to package this as `verapdf`. Any reason we shouldn't just use that name instead?

Comment (alexreg) [reply Formula/verapdf-apps.rb:1]: @carlocab I'm not particularly bothered to be honest, I just thought that if a [`verapdf-library`](https://github.com/veraPDF/veraPDF-library) package is created in the future, there might be a name clash. Up to you, of course.

Comment (alexreg) [reply Formula/verapdf-apps.rb:1]: @carlocab Just fished an issue with livecheck. Let me know about the above and then we can get this PR merged, I presume.

**Comment by alexreg** [comment]
I believe this is ready for merge now, unless a rename or alias is desired, re @carlocab's comment above.

File: Formula/verapdf-apps.rb
Comment (carlocab) [reply Formula/verapdf-apps.rb:1]: Is this the same as what you get when you install the software from [this page](https://verapdf.org/software/)? If so, we should probably call it `verapdf`.

File: Formula/verapdf-apps.rb
Comment (carlocab) [inline Formula/verapdf-apps.rb:34]: Did this need changing?

Comment (alexreg) [reply Formula/verapdf-apps.rb:34]: Yes, sorry, I forgot to revert after briefly testing something.

**Comment by alexreg** [comment]
@carlocab Thanks for the review. The main binary is named `verapdf`, so we'll rename the formula to that, I suppose. If a formula for the library is created in the future, it can be called `verapdf-lib` or something similar, I suppose.

**Review by carlocab — COMMENTED** [review]
The filename needs updating to `verapdf`.

**Review by carlocab — APPROVED** [review]
Thanks for your patience! The formula looks to have turned out quite well.

**Comment by alexreg** [comment]
@carlocab Yeah, it was pretty straightforward after getting some stuff merged in upstream. Thanks for accepting.

**Pull request closed without merging**
